test/grizzly/zwave/commands/thermostat_mode_supported_report_test.exs

```diff
@@ -0,0 +1,102 @@
+defmodule Grizzly.ZWave.Commands.ThermostatModeSupportedReportTest do
+  use ExUnit.Case, async: true
+
+  alias Grizzly.ZWave.Commands.ThermostatModeSupportedReport
+
+  test "creates the command and validates params" do
+    {:ok, _command} = ThermostatModeSupportedReport.new(modes: [])
+
+    modes = [
+      off: true,
+      heat: true,
+      cool: true,
+      auto: true,
+      auxiliary: true,
+      resume_on: true,
+      fan: true,
+      furnace: true,
+      dry: true,
+      moist: true,
+      auto_changeover: true,
+      energy_heat: true,
+      energy_cool: true,
+      away: true,
+      full_power: true,
+      manufacturer_specific: true
+    ]
+
+    {:ok, _command} = ThermostatModeSupportedReport.new(modes: modes)
+  end
+
+  test "encodes params correctly" do
+    {:ok, command} = ThermostatModeSupportedReport.new(modes: [])
+    assert <<>> = ThermostatModeSupportedReport.encode_params(command)
+
+    modes = [
+      off: true,
+      heat: true,
+      cool: true,
+      auto: true,
+      auxiliary: true,
+      resume_on: false,
+      fan: false,
+      furnace: false,
+      dry: false,
+      moist: false,
+      auto_changeover: false,
+      energy_heat: true,
+      energy_cool: true,
+      away: false,
+      full_power: false,
+      manufacturer_specific: false
+    ]
+
+    {:ok, command} = ThermostatModeSupportedReport.new(modes: modes)
+    assert <<0x1F, 0x18>> = ThermostatModeSupportedReport.encode_params(command)
+
+    modes = [
+      off: true,
+      heat: true,
+      cool: true,
+      auto: true,
+      auxiliary: true,
+      resume_on: true,
+      fan: true,
+      furnace: true,
+      dry: true,
+      moist: true,
+      auto_changeover: true,
+      energy_heat: true,
+      energy_cool: true,
+      away: true,
+      full_power: true,
```

Comment:
Potentially stupid question: The spec states the value at this index is reserved and the next is `full_power`. Do we have to convey that at all here?

Reply:
For this style of bitmask, the flags are indexed by their value, not by their position in the list. Full power is [0x0F](https://github.com/smartrent/grizzly/blob/main/lib/grizzly/zwave/command_classes/thermostat_mode.ex#L51), and away is 0x0D, leaving a gap for the reserved index.